### PR TITLE
S16 show rhythm codes cont - Christina/Irianni/Man/Delnia

### PR DESCRIPTION
**Pull request by christina-mifsud**
## Description

Implemented functionality for rhythm code icons - number of icons shown reflects number of rhythm codes, on click an api call is made which fetches corresponding rhythms from db to populates drop down list. When list item is clicked, the modal shows with corresponding video/audio/text.

Styling of rhythm code icons and rhythms pane.

Addresses #16 and #44

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have carefully reviewed my own code
- [x] I have commented my code
- [x] I have updated any documentation

### CODE REVIEW

File: client/src/App.js [hunk 1]
```diff
@@ -17,8 +17,10 @@ const App = () => (
 		<Route path="/ModalTestingPage" element={<ModalTestingPage />} />
 		<Route path="/" element={<Home />} />
 		<Route path="/about/this/site" element={<About />} />
-		<Route path="/RhythmsPanelTestingPage" element={<RhythmsPanelTestingPage />}
-			/></Routes>
+		<Route path="/testingpage" element={<ModalTestingPage />} />
```
Comment (mferryRV) [inline client/src/App.js]: Not sure if needed?

File: client/src/pages/RhythmsPanelTestingPage.js
Comment (mferryRV) [inline client/src/pages/RhythmsPanelTestingPage.js:1]: Delete before merge

File: client/src/pages/Home.js [hunk 2]
```diff
@@ -2,79 +2,86 @@ import { useState, useEffect } from "react";
 import { Link } from "react-router-dom";
 
 import "./Home.css";
-import ModalVideo from "../components/ModalVideo";
 import CityIcon from "./CityIcon";
 import DotMap from "../components/DotMap";
+import ModalVideo from "../components/ModalVideo";
 
-export function Home() {
+function MyMap() {
```
Comment (mferryRV) [inline client/src/pages/Home.js]: Needs to be in components folder and imported to Home.js

File: client/src/pages/Home.js [hunk 3]
```diff
@@ -2,79 +2,86 @@ import { useState, useEffect } from "react";
...
-				<br></br>
-				<Link to="/ModalTestingPage">Modal Testing</Link>
-				<br></br>
-				<Link to="/SliderTest">slider test</Link>
-				<br></br>
-				<Link to="/about/this/site">About</Link>
-				<br></br>
-				<Link to="/RhythmsPanelTestingPage">RhythmsPanelTestingPage</Link>
+			<MyMap />
+			<br />
```
Comment (mferryRV) [inline client/src/pages/Home.js]: OK but let's style in the final version

File: client/src/pages/Home.js [hunk 4]
```diff
@@ -2,79 +2,86 @@ import { useState, useEffect } from "react";
 import { Link } from "react-router-dom";
 
 import "./Home.css";
-import ModalVideo from "../components/ModalVideo";
 import CityIcon from "./CityIcon";
 import DotMap from "../components/DotMap";
+import ModalVideo from "../components/ModalVideo";
```
Comment (mferryRV) [inline client/src/pages/Home.js]: Move this back to line 5 to avoid conflicts

File: client/src/pages/Home.js [hunk 5]
```diff
@@ -2,79 +2,86 @@ import { useState, useEffect } from "react";
...
 
-    const [rhythms, setRhythms] = useState([]);
+	const API_Rhythms_URL = "/api/rhythms";
 
-		const API_URL = "/api/rhythm_codes";
+	useEffect(() => {
+		fetch(API_Rhythms_URL)
+			.then((response) => response.json())
+			.then((rhythms) => {
+				console.log("API resultsssss:", rhythms);
```
Comment (mferryRV) [inline client/src/pages/Home.js]: No console logs in production

File: client/src/pages/Home.js [hunk 6]
```diff
@@ -2,79 +2,86 @@ import { useState, useEffect } from "react";
...
+					if (!existingRhythm) {
+						acc.push({
+							rhythm_code: rhythm.rhythm_code,
+							leftpx: rhythm.leftpx,
+							toppx: rhythm.toppx,
+						});
+					}
+					return acc;
+				}, []);
+				console.log(rhythmsCodes);
```
Comment (mferryRV) [inline client/src/pages/Home.js]: This too

File: client/src/pages/Home.js [hunk 7]
```diff
@@ -2,79 +2,86 @@ import { useState, useEffect } from "react";
...
+			})
+			.catch((error) => {
+				console.error("NOPE! Rhythms:", error);
+			});
+	}, []);
 
+	return (
+		<div>
+				{rhythmCodes.map((rhythmCodeObject) => {
+					return (
```
Comment (mferryRV) [inline client/src/pages/Home.js]: For brevity I'd recommend removing the return statement and doing:
```js
rhythmCodes.map((rhythmCodeObject => <CityIcon />)
```

File: client/src/pages/Home.js [hunk 8]
```diff
@@ -2,79 +2,86 @@ import { useState, useEffect } from "react";
...
-		setShow(true);
-		setSelectedIcon(videoInfo); // useState to store the selected video info
-	};
+						/>
+					);
+				})}
+				{isModalOpen &&
+					<ModalVideo setModalOpen={setModalOpen} rhythm={selectedRhythm} />
+				}
+			<DotMap className="map" />
```
Comment (mferryRV) [inline client/src/pages/Home.js]: DotMap should go above, as it is effectively underneath the other components

File: client/src/pages/Home.js [hunk 9]
```diff
@@ -2,79 +2,86 @@ import { useState, useEffect } from "react";
...
-// dropdown and useEffect to fetch from rhythms
-		setShow(true);
-		setSelectedIcon(videoInfo); // useState to store the selected video info
-	};
+						/>
+					);
+				})}
+				{isModalOpen &&
+					<ModalVideo setModalOpen={setModalOpen} rhythm={selectedRhythm} />
+				}
```
Comment (mferryRV) [inline client/src/pages/Home.js]: Please remove one indent - sorry, not sure why prettier isn't pretty

File: client/src/pages/Home.css [hunk 10]
```diff
@@ -1,10 +1,29 @@
...
+  padding: 0;
+  box-sizing: border-box;
+}
+
+:root {
+  --font-roboto: "Roboto", sans-serif;
+  --copy-color: #CB3100;
+}
+
+/* .App {
```
Comment (mferryRV) [inline client/src/pages/Home.css]: Remove commented CSS

Comment (mferryRV) [reply client/src/pages/Home.css]: .App should probably be in App.css or similar?

File: client/src/pages/Home.css [hunk 11]
```diff
@@ -1,10 +1,29 @@
+  @import url('https://fonts.googleapis.com/css2?family=Roboto:wght@400;500;700;900&display=swap');
+* {
```
Comment (mferryRV) [inline client/src/pages/Home.css:2]: This stuff should be at the top level, like App.css - do that in another PR if you want

File: client/src/pages/CityIcon.js [hunk 12]
```diff
@@ -1,20 +1,71 @@
+import { useState } from "react";
```
Comment (mferryRV) [inline client/src/pages/CityIcon.js]: Needs to move into components folder

File: client/src/pages/CityIcon.js [hunk 13]
```diff
@@ -1,20 +1,71 @@
...
+// const style = (toppx, leftpx) => ({
+// 	position: "absolute",
+// 	top: `${toppx}`,
+// 	left: `${leftpx}px`,
+// 	width: "40",
+// 	height:"40",
+// 	viewBox:"0 0 40 40",
+// 	fill:"none",
+
+// });
```
Comment (mferryRV) [inline client/src/pages/CityIcon.js]: Delete commented code

File: client/src/pages/CityIcon.js [hunk 14]
```diff
@@ -1,20 +1,71 @@
...
+
+function CityIcon({ toppx, leftpx, rhythmCodeName, rhythmsList, setModalOpen, setSelectedRhythm }) {
+	console.log(`toppx: ${toppx}, leftpx: ${leftpx}`);
```
Comment (mferryRV) [inline client/src/pages/CityIcon.js]: No console logs

File: client/src/pages/Home.css [hunk 15]
```diff
@@ -13,48 +32,47 @@
...
-	height:20px; 
-}  
-
-.dropdown {
+/* .dropdown {
   position: relative;
+  background-color: aqua;
+} */
+
+.rhythmCodeIcon{
```
Comment (mferryRV) [inline client/src/pages/Home.css]: Move this into CityIcon.css

File: client/src/pages/Home.css [hunk 16]
```diff
@@ -13,48 +32,47 @@
...
+  width:40px;
+	height:40px;
 }
 
-.menu {
+.rhythmItemList {
```
Comment (mferryRV) [inline client/src/pages/Home.css]: Move this into RhythmItem.css

File: client/src/pages/Home.css [hunk 17]
```diff
@@ -13,48 +32,47 @@
...
 
-.menu > li {
-  margin: 0;
-  background-color: white;
-}
-
-.menu > li:hover {
-  background-color: rgb(255, 236, 167);
-}
+.rhythmCodeName {
```
Comment (mferryRV) [inline client/src/pages/Home.css]: CityIcon.css

File: client/src/pages/Home.css [hunk 18]
```diff
@@ -13,48 +32,47 @@
...
+  font-weight: 700;
+  list-style-type: none;
+  margin: 5px 0;
+  padding-bottom: 8px;
+} 
 
-.menu > li > button {
-  width: 100%;
-  height: 100%;
+.rhythmItemList > li > button {
```
Comment (mferryRV) [inline client/src/pages/Home.css]: This and below go in RhythmItem.css

**Review by mferryRV — COMMENTED** [review]
Pretty much done - all JS looks good, all styling looks good, basically just refactoring the file structure and removing comments then it's good to go

**Comment by christina-mifsud** [comment]
Changes made - branch ready to be merged! Thanks! 
Screenshot from 2023-10-15 21-02-46

File: client/src/App.js [hunk 19]
```diff
@@ -4,10 +4,6 @@ import ModalTestingPage from "./pages/ModalTestingPage";
 import Home from "./pages/Home";
 import MapSection from "./components/MapSection";
 import SliderTest from "./pages/SliderTest";
-import DataFetchingTest from "./pages/DataFetchingTest";
-
-import RhythmsPanelTestingPage from "./pages/RhythmsPanelTestingPage";
```
Comment (mferryRV) [inline client/src/App.js:8]: Will it run without this component? This is used on line 18

File: client/src/App.js
Comment (mferryRV) [inline client/src/App.js:1]: Need to remove `/RhythmsPanelTestingPage` from the routes

File: client/src/components/ModalVideo.js [hunk 20]
```diff
@@ -23,7 +15,7 @@ function ModalVideo({
 
 	let modalContent;
 
-	if (url && description) {
+	if (rhythm.url && rhythm.description) {
```
Comment (mferryRV) [inline client/src/components/ModalVideo.js]: `rhythm.url` needs to reference `rhythm.video` as this is what the API returns

File: client/src/components/ModalVideo.js [hunk 21]
```diff
@@ -53,35 +45,30 @@ function ModalVideo({
...
 	} else {
 		modalContent = (
 			<div>
-				{url && <VideoCard url={url} />}
+				{rhythm.url && <VideoCard url={rhythm.url} />}
 				{audioUrl && (
-					<CustomAudioPlayer audioUrl={audioUrl} />
+					<CustomAudioPlayer audioUrl={rhythm.audioUrl} />
 				)}
 				{description && <div className="text">{description}</div>}
```
Comment (mferryRV) [inline client/src/components/ModalVideo.js]: description needs to reference rhythm.description

File: client/src/components/ModalVideo.js [hunk 22]
```diff
@@ -53,35 +45,30 @@ function ModalVideo({
...
 					</Tab.Pane>
 				</Tab.Content>
 			</Tab.Container>
 		);
 	} else {
 		modalContent = (
 			<div>
-				{url && <VideoCard url={url} />}
+				{rhythm.url && <VideoCard url={rhythm.url} />}
 				{audioUrl && (
```
Comment (mferryRV) [inline client/src/components/ModalVideo.js]: `rhythm.audio` replaces `audioUrl`

**Review by mferryRV — COMMENTED** [review]
I tried running this locally and found two issues:
1. Component deleted but the route isn't deleted from `App.js`
1. ModalVideo references old versions of the rhythm object - I've added comments for `rhythm.url` to `rhythm.video`, `audio` to `rhythm.audio` and `description` to `rhythm.description`

**Review by mferryRV — APPROVED** [review]
🎉
